ui-v2/app/helpers/href-mut.js

```diff
@@ -0,0 +1,39 @@
+import Helper from '@ember/component/helper';
+import { inject as service } from '@ember/service';
+import { hrefTo } from 'consul-ui/helpers/href-to';
+
+const getRouteParams = function(route, params = {}) {
+  return route.paramNames.map(function(item) {
+    if (typeof params[item] !== 'undefined') {
+      return params[item];
+    }
+    return route.params[item];
+  });
+};
+export default Helper.extend({
+  router: service('router'),
+  compute([params], hash) {
+    let current = this.router.currentRoute;
+    let parent;
+    let atts = getRouteParams(current, params);
+    // walk up the entire route/s replacing any instances
+    // of the specified params with the values specified
+    while ((parent = current.parent)) {
+      atts = atts.concat(getRouteParams(parent, params));
+      current = parent;
+    }
+    let route = this.router.currentRoute.name;
+    // TODO: this is specific to consul/nspaces
+    // 'ideally' we could try and do this elsewhere
+    // not super important though.
+    // This will turn an URL that has no nspace (/ui/dc-1/nodes) into one
+    // that does have a namespace (/ui/~nspace/nodes) if you href-mut with
```

Comment:
is this right that the nspace replaces the datacenter in the route? (sorry I know that this is a comment, but I wasn't sure of the interplay of namespaces and datacenters).

Reply:
Somewhat related: I know nested namespaces aren't supported at this time - is that expected in the future? will the format then be with slashes and url encoding will come into play here (or somewhere around here)?

Reply:
Ah good spot, theres a typo here which makes the comment confusing in this context! Will tweak, thanks!

>  I know nested namespaces aren't supported at this time - is that expected in the future?

I think its expected yeah, but I think we covered this future potential in RFC-land (we'll either encode them or switch them out for another non-DNS character)